### PR TITLE
machines: input validation errors should appear inline

### LINKED ISSUE #11065: VM name error is disconnected from it's input form
Cockpit version: master
OS: Fedora 29
Page: Machines

The error I get when I didn't fill the name field is very far away from the field itself. It takes some unnecessary cognitive work to locate the name field and fix it.

Steps to reproduce
1. Create a new name
2. Leave the name field blank

screenshot from 2019-01-28 13-41-24

### PR DESCRIPTION
**Pull request by KKoukiou**
Adjust create VM dialog to support inline validation errors.

Also get rid of validation for negative values for memory and
storage size since UI does not allow these anyway.

TestCreate was adjusted to search for inline validation errors.

Fixes #11065

### CODE REVIEW

**Comment by KKoukiou** [comment]
create-dialog-inline

@andreasn  tbh I don't like verbs like please in the dialogs :_) But if you think it's appropriate I 'll have to keep the errors consistent and change them both. What do you think?

Also otherwise it looks ok? I had to overwrite the form-group class margins btw.

File: test/verify/machineslib.py [hunk 1]
```diff
@@ -1343,32 +1364,28 @@ def allowBugErrors(location, original_exception):
...
-                            allowBugErrors(error_location, x2)
+            b.wait_present(".modal-footer .spinner")
+            b.wait_not_present(".modal-footer .spinner")
+            try:
+                with b.wait_timeout(10):
+                    b.wait_present(error_location)
+                    waitForError(errors, error_location)
+
+                # dialog can complete if the error was not returned immediately
+            except Exception as x1:
```
Comment (martinpitt) [inline test/verify/machineslib.py]: Please catch `testlib.Error` here, not a blanket exception. (Once more below)

Comment (KKoukiou) [reply test/verify/machineslib.py]: Oh that was just huge  whitespace change. Didn't introduce this here.
But I 'll post a commit on top of this one to fix this all these plan Exception catches in this file.

**Comment by KKoukiou** [comment]
@martinpitt  It would be good idea to stick to one style for showing inline errors in dialogs. At least for pages using React.
@andreasn can we standardize that?

**Comment by andreasn** [comment]
> I don't like verbs like please in the dialogs :_) But if you think it's appropriate I 'll have to keep the errors consistent and change them both. What do you think?

I don't have strong opinions on this, so omitting _please_ is fine by me. Looks good to go!

File: pkg/machines/components/create-vm-dialog/createVmDialog.less [hunk 2]
```diff
@@ -15,3 +15,11 @@
 #system-select .dropdown-menu {
     max-height: ~"calc(90vh - 50px - 10em)";
 }
+
+
+/* We use form-group in divs for inline validaton of dialog values
+ * and we don't need extra margins there.
+ */
+#create-vm-dialog div.form-group {
+    margin-bottom: 0 !important;
```
Comment (andreasn) [inline pkg/machines/components/create-vm-dialog/createVmDialog.less]: If you can figure out a way to not use !important here, I think that would be healthier for the code long term. Maybe by specifying the selector a bit closer?

Comment (andreasn) [reply pkg/machines/components/create-vm-dialog/createVmDialog.less]: @garrett any good ideas for how to solve this without the !important?

Comment (KKoukiou) [reply pkg/machines/components/create-vm-dialog/createVmDialog.less]: I fixed by just specifying a more specific selector.

Comment (andreasn) [reply pkg/machines/components/create-vm-dialog/createVmDialog.less]: Awesome, thanks!

**Pull request merged**